specs/protocol/mpt-migration/exec-engine.md

```diff
@@ -0,0 +1,102 @@
+  - [Transition from ZK Trie to Merkle Patricia Trie](#transition-from-zk-trie-to-merkle-patricia-trie)
+  - [`SELFDESTRUCT` opcode](#selfdestruct-opcode)
+  - [`isSystemTransaction` field in `DepositTx` type](#issystemtransaction-field-in-deposittx-type)
+- [Fee Distribution Process](#fee-distribution-process)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+[g-zktrie]: ../../glossary.md#zk-trie
+[g-mpt]: ../../glossary.md#merkle-patricia-trie
+[validator-system]: ../validator-v2/overview.md
+
+## Overview
+
+This upgrade changes the underlying trie structure of the execution client from the [ZK Trie][g-zktrie] to the
+[Merkle Patricia Trie][g-mpt]. There are also subsequent changes to the EVM to enhance the compatibility with
+Vanilla OP Stack.
+
+## Timestamp Activation
+
+Kroma MPT Migration upgrade, like other network upgrades, is activated at a timestamp. Changes to the L2 Block execution
+rules are applied when the `L2 Timestamp >= activation time`.
+
+## Historical RPC
+
+Since the underlying trie structure is changed from the ZK Trie to the Merkle Patricia Trie, the historical RPC is used
+for the RPC calls that are related to the historical data before the Kroma MPT Migration upgrade. The historical RPC
+provides the historical data that is stored in the ZK Trie. The following endpoints will route the call to the
+historical RPC if the requested timestamp is before the activation time of the Kroma MPT Migration upgrade:
+
+### `eth_` namespace
+
+- `eth_getBalance`
+- `eth_getProof`
+- `eth_getCode`
+- `eth_getStorageAt`
+- `eth_call`
+- `eth_estimateGas`
+- `eth_createAccessList`
+- `eth_getTransactionCount`
+
+### `debug_` namespace
+
+- `debug_traceBlockByNumber`
+- `debug_traceBlockByHash`
+- `debug_traceTransaction`
+
+## EVM Changes
+
+### Transition from ZK Trie to Merkle Patricia Trie
+
+The underlying trie structure of the execution client is changed from the ZK Trie to the Merkle Patricia Trie. This
+enhances the performance of the execution client, and reduces the operational cost of maintaining zkEVM circuits by
+enabling the proving scheme transition to zkVM.
+
+### `SELFDESTRUCT` opcode
+
+The `SELFDESTRUCT` opcode was regarded as invalid opcode, since zkEVM proving system used in ZK fault proof cannot prove
+the self-destruct operation. However, with the transition to zkVM, the `SELFDESTRUCT` opcode is now enabled, following
+rules that are defined at [EIP-6780].
+
+[EIP-6780]: https://eips.ethereum.org/EIPS/eip-6780
+
+### `isSystemTransaction` field in `DepositTx` type
+
+In the previous version of the protocol, the `isSystemTransaction` boolean was removed. To enhance the compatibility
+with vanilla OP Stack, the `isSystemTransaction` boolean is re-enabled.
+
+However, since `isSystemTransaction` was disabled as in the [Regolith upgrade](../regolith/overview.md), system
+transactions remain to use the same gas accounting rules as regular deposits.
+
+## Fee Distribution Process
```

Comment:
Isn't it included in EVM Changes section?

Reply:
Agreed, I'll move it. Thanks!